glad results in undefined symbols on Mac OS for simple example

If I replace the [libigl example project](https://github.com/libigl/libigl-example-project)'s main.cpp with this minimal example

```
#include <igl/opengl/create_shader_program.h>

int main()
{
  GLuint Fg_prog_id = igl::opengl::create_shader_program("","",{});
}
```

I get the following undefined symbol errors on Mac OS X:

```
[ 95%] Linking CXX executable example_bin
/usr/local/Cellar/cmake/3.10.2/bin/cmake -E cmake_link_script CMakeFiles/example_bin.dir/link.txt --verbose=1
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++  -O3 -DNDEBUG -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.13.sdk -Wl,-search_paths_first -Wl,-headerpad_max_install_names  CMakeFiles/example_bin.dir/main.cpp.o  -o example_bin -framework OpenGL glad/libglad.a glfw/src/libglfw3.a -framework Cocoa -framework IOKit -framework CoreFoundation -framework CoreVideo 
Undefined symbols for architecture x86_64:
  "_glad_glAttachShader", referenced from:
      igl::opengl::create_shader_program(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::map<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, unsigned int, std::__1::less<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >, std::__1::allocator<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const, unsigned int> > > const&, unsigned int&) in main.cpp.o
  "_glad_glBindAttribLocation", referenced from:
      igl::opengl::create_shader_program(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::map<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, unsigned int, std::__1::less<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >, std::__1::allocator<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const, unsigned int> > > const&, unsigned int&) in main.cpp.o
  "_glad_glCompileShader", referenced from:
      igl::opengl::load_shader(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, unsigned int) in main.cpp.o
  "_glad_glCreateProgram", referenced from:
      igl::opengl::create_shader_program(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::map<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, unsigned int, std::__1::less<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >, std::__1::allocator<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const, unsigned int> > > const&, unsigned int&) in main.cpp.o
  "_glad_glCreateShader", referenced from:
      igl::opengl::load_shader(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, unsigned int) in main.cpp.o
  "_glad_glDeleteShader", referenced from:
      igl::opengl::create_shader_program(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::map<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, unsigned int, std::__1::less<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >, std::__1::allocator<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const, unsigned int> > > const&, unsigned int&) in main.cpp.o
  "_glad_glDetachShader", referenced from:
      igl::opengl::create_shader_program(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::map<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, unsigned int, std::__1::less<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >, std::__1::allocator<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const, unsigned int> > > const&, unsigned int&) in main.cpp.o
  "_glad_glGetProgramInfoLog", referenced from:
      igl::opengl::create_shader_program(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::map<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, unsigned int, std::__1::less<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >, std::__1::allocator<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const, unsigned int> > > const&, unsigned int&) in main.cpp.o
  "_glad_glGetProgramiv", referenced from:
      igl::opengl::create_shader_program(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::map<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, unsigned int, std::__1::less<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >, std::__1::allocator<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const, unsigned int> > > const&, unsigned int&) in main.cpp.o
  "_glad_glGetShaderInfoLog", referenced from:
      igl::opengl::load_shader(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, unsigned int) in main.cpp.o
  "_glad_glGetShaderiv", referenced from:
      igl::opengl::load_shader(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, unsigned int) in main.cpp.o
  "_glad_glLinkProgram", referenced from:
      igl::opengl::create_shader_program(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::map<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, unsigned int, std::__1::less<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >, std::__1::allocator<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const, unsigned int> > > const&, unsigned int&) in main.cpp.o
  "_glad_glShaderSource", referenced from:
      igl::opengl::load_shader(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, unsigned int) in main.cpp.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [example_bin] Error 1
make[1]: *** [CMakeFiles/example_bin.dir/all] Error 2
make: *** [all] Error 2
```

This is strange because `libglad.a` contains these symbols  (revealed via `nm`) and the vanilla example `main.cpp` calls the Viewer which in turn calls MeshGL which calls `create_shader_program`, and links just fine. I haven't tested yet if this is happening on other build environments. I'm using 

```
Apple LLVM version 9.1.0 (clang-902.0.39.1)
Target: x86_64-apple-darwin17.5.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```
on Mac OS 10.13 and used  `cmake ../ -DLIBIGL_USE_STATIC_LIBRARY=Off -DCMAKE_BUILD_TYPE=Release` to create the build.

--- a/cmake/recipes/external/glad.cmake
+++ b/cmake/recipes/external/glad.cmake
@@ -8,7 +8,7 @@ include(FetchContent)
 FetchContent_Declare(
     glad
     GIT_REPOSITORY https://github.com/libigl/libigl-glad.git
-    GIT_TAG        09b4969c56779f7ddf8e6176ec1873184aec890f
+    GIT_TAG        ceef55fcd08bdd16e985370a99cfb60e69623221
 )
 
 FetchContent_MakeAvailable(glad)

--- a/include/igl/opengl/gl.h
+++ b/include/igl/opengl/gl.h
@@ -20,6 +20,6 @@
 //     #include <GL/gl.h>
 //
 
-#include <glad/glad.h>
+#include <glad/gl.h>
 
 #endif

--- a/include/igl/opengl/glfw/Viewer.cpp
+++ b/include/igl/opengl/glfw/Viewer.cpp
@@ -177,7 +177,7 @@ namespace glfw
     }
     glfwMakeContextCurrent(window);
     // Load OpenGL and its extensions
-    if (!gladLoadGLLoader((GLADloadproc) glfwGetProcAddress))
+    if (!gladLoadGL((GLADloadfunc) glfwGetProcAddress))
     {
       printf("Failed to load OpenGL and its extensions\n");
       return(-1);

--- a/include/igl/opengl/glfw/background_window.cpp
+++ b/include/igl/opengl/glfw/background_window.cpp
@@ -17,7 +17,7 @@ IGL_INLINE bool igl::opengl::glfw::background_window(GLFWwindow* & window)
   window = glfwCreateWindow(1, 1,"", NULL, NULL);
   if(!window) return false;
   glfwMakeContextCurrent(window);
-  if (!gladLoadGLLoader((GLADloadproc) glfwGetProcAddress))
+  if (!gladLoadGL((GLADloadfunc) glfwGetProcAddress))
   {
     printf("Failed to load OpenGL and its extensions");
   }

